### ISSUE TITLE
how to save the results when finish running the command " python3 deepdefense.py --pretest --dataset mnist --arch LeNet"

### ISSUE THREAD

**Issue by aspnetcs** [ZiangYan/deepdefense.pytorch#5]
how to save the results when finish running the command " python3 deepdefense.py --pretest --dataset mnist --arch LeNet"

**Comment by ZiangYan** [comment]
Hi, all models will be saved in ```args.exp_dir```.

**Comment by aspnetcs** [comment]
我是说输出到屏幕上的中间结果，用于做曲线图的，如下所示。
......
I0312 12:07:00.613735 18 deepdefense.py:486]    ratio           : 0.239105
I0312 12:07:00.613817 18 deepdefense.py:487]    ce_loss         : 0.003888
I0312 12:07:00.613897 18 deepdefense.py:488]    loss            : 0.387850
I0312 12:07:00.613976 18 deepdefense.py:489]    accuracy        : 1.000000
I0312 12:07:00.692862 18 deepdefense.py:483] Processing 2600 - 2700 / 60000
I0312 12:07:00.693001 18 deepdefense.py:484]    noise_norm      : 409.451950
I0312 12:07:00.693105 18 deepdefense.py:485]    grad_norm       : 24.116674
I0312 12:07:00.693203 18 deepdefense.py:486]    ratio           : 0.224275
I0312 12:07:00.693285 18 deepdefense.py:487]    ce_loss         : 0.020267
I0312 12:07:00.693365 18 deepdefense.py:488]    loss            : 0.394605
I0312 12:07:00.693444 18 deepdefense.py:489]    accuracy        : 0.990000
I0312 12:07:00.766579 18 deepdefense.py:483] Processing 2700 - 2800 / 60000
I0312 12:07:00.766695 18 deepdefense.py:484]    noise_norm      : 424.162316
I0312 12:07:00.766824 18 deepdefense.py:485]    grad_norm       : 28.656344
I0312 12:07:00.766938 18 deepdefense.py:486]    ratio           : 0.227037
I0312 12:07:00.767028 18 deepdefense.py:487]    ce_loss         : 0.009213
I0312 12:07:00.767117 18 deepdefense.py:488]    loss            : 0.559985
I0312 12:07:00.767210 18 deepdefense.py:489]    accuracy        : 1.000000
I0312 12:07:00.846417 18 deepdefense.py:483] Processing 2800 - 2900 / 60000
I0312 12:07:00.846539 18 deepdefense.py:484]    noise_norm      : 441.390101
I0312 12:07:00.846654 18 deepdefense.py:485]    grad_norm       : 24.307082
I0312 12:07:00.846759 18 deepdefense.py:486]    ratio           : 0.238342
I0312 12:07:00.846866 18 deepdefense.py:487]    ce_loss         : 0.029882
I0312 12:07:00.846957 18 deepdefense.py:488]    loss            : 0.309963
I0312 12:07:00.847044 18 deepdefense.py:489]    accuracy        : 0.990000
I0312 12:07:00.931418 18 deepdefense.py:483] Processing 2900 - 3000 / 60000
I0312 12:07:00.931542 18 deepdefense.py:484]    noise_norm      : 391.050166
I0312 12:07:00.931686 18 deepdefense.py:485]    grad_norm       : 20.951988
I0312 12:07:00.931795 18 deepdefense.py:486]    ratio           : 0.210398
I0312 12:07:00.931886 18 deepdefense.py:487]    ce_loss         : 0.064489
I0312 12:07:00.931974 18 deepdefense.py:488]    loss            : 0.415758
I0312 12:07:00.932064 18 deepdefense.py:489]    accuracy        : 0.980000
.....
这些结果如何保存？以便用于作图。










--




At 2021-03-12 09:25:41, "yanziang" ***@***.***> wrote:

Hi, all models will be saved in args.exp_dir.

**Comment by aspnetcs** [comment]
dataset, victim models, and pre-trained weights of policy networks in google driver (pda.pytorch ),能够把这些数据放到百度云盘中吗？应为在中国，不翻墙，无法访问google 云盘。










--




At 2021-03-12 09:25:41, "yanziang" ***@***.***> wrote:

Hi, all models will be saved in args.exp_dir.

**Comment by ZiangYan** [comment]
> 我是说输出到屏幕上的中间结果，用于做曲线图的，如下所示。 ...... I0312 12:07:00.613735 18 deepdefense.py:486] ratio : 0.239105 I0312 12:07:00.613817 18 deepdefense.py:487] ce_loss : 0.003888 I0312 12:07:00.613897 18 deepdefense.py:488] loss : 0.387850 I0312 12:07:00.613976 18 deepdefense.py:489] accuracy : 1.000000 I0312 12:07:00.692862 18 deepdefense.py:483] Processing 2600 - 2700 / 60000 I0312 12:07:00.693001 18 deepdefense.py:484] noise_norm : 409.451950 I0312 12:07:00.693105 18 deepdefense.py:485] grad_norm : 24.116674 I0312 12:07:00.693203 18 deepdefense.py:486] ratio : 0.224275 I0312 12:07:00.693285 18 deepdefense.py:487] ce_loss : 0.020267 I0312 12:07:00.693365 18 deepdefense.py:488] loss : 0.394605 I0312 12:07:00.693444 18 deepdefense.py:489] accuracy : 0.990000 I0312 12:07:00.766579 18 deepdefense.py:483] Processing 2700 - 2800 / 60000 I0312 12:07:00.766695 18 deepdefense.py:484] noise_norm : 424.162316 I0312 12:07:00.766824 18 deepdefense.py:485] grad_norm : 28.656344 I0312 12:07:00.766938 18 deepdefense.py:486] ratio : 0.227037 I0312 12:07:00.767028 18 deepdefense.py:487] ce_loss : 0.009213 I0312 12:07:00.767117 18 deepdefense.py:488] loss : 0.559985 I0312 12:07:00.767210 18 deepdefense.py:489] accuracy : 1.000000 I0312 12:07:00.846417 18 deepdefense.py:483] Processing 2800 - 2900 / 60000 I0312 12:07:00.846539 18 deepdefense.py:484] noise_norm : 441.390101 I0312 12:07:00.846654 18 deepdefense.py:485] grad_norm : 24.307082 I0312 12:07:00.846759 18 deepdefense.py:486] ratio : 0.238342 I0312 12:07:00.846866 18 deepdefense.py:487] ce_loss : 0.029882 I0312 12:07:00.846957 18 deepdefense.py:488] loss : 0.309963 I0312 12:07:00.847044 18 deepdefense.py:489] accuracy : 0.990000 I0312 12:07:00.931418 18 deepdefense.py:483] Processing 2900 - 3000 / 60000 I0312 12:07:00.931542 18 deepdefense.py:484] noise_norm : 391.050166 I0312 12:07:00.931686 18 deepdefense.py:485] grad_norm : 20.951988 I0312 12:07:00.931795 18 deepdefense.py:486] ratio : 0.210398 I0312 12:07:00.931886 18 deepdefense.py:487] ce_loss : 0.064489 I0312 12:07:00.931974 18 deepdefense.py:488] loss : 0.415758 I0312 12:07:00.932064 18 deepdefense.py:489] accuracy : 0.980000 ..... 这些结果如何保存？以便用于作图。
> […](#)
> -- At 2021-03-12 09:25:41, "yanziang" ***@***.***> wrote: Hi, all models will be saved in args.exp_dir. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

You can try to use ```>``` (shell) command to redirect stdout and stderr.

**Comment by ZiangYan** [comment]
> dataset, victim models, and pre-trained weights of policy networks in google driver (pda.pytorch ),能够把这些数据放到百度云盘中吗？应为在中国，不翻墙，无法访问google 云盘。
> […](#)
> -- At 2021-03-12 09:25:41, "yanziang" ***@***.***> wrote: Hi, all models will be saved in args.exp_dir. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

No. It's hard to upload and download large file from BaiduNetdisk without a so-called baidu vvip membership account.

I'm afraid that I cannot provide share link via BaiduNetdisk.

Sorry for that.

Moreover, please open issues in pda.pytorch repo to discuss about pda related things.

**Comment by aspnetcs** [comment]
python3 deepdefense.py --pretest --dataset mnist --arch LeNet >> results.txt
不管用的，还是输出到屏幕。我试过很多次了。你看如附件中的运行截图。












--




在 2021-03-12 21:57:58，"yanziang" ***@***.***> 写道：

我是说输出到屏幕上的中间结果，用于做曲线图的，如下所示。 ...... I0312 12:07:00.613735 18 deepdefense.py:486] ratio : 0.239105 I0312 12:07:00.613817 18 deepdefense.py:487] ce_loss : 0.003888 I0312 12:07:00.613897 18 deepdefense.py:488] loss : 0.387850 I0312 12:07:00.613976 18 deepdefense.py:489] accuracy : 1.000000 I0312 12:07:00.692862 18 deepdefense.py:483] Processing 2600 - 2700 / 60000 I0312 12:07:00.693001 18 deepdefense.py:484] noise_norm : 409.451950 I0312 12:07:00.693105 18 deepdefense.py:485] grad_norm : 24.116674 I0312 12:07:00.693203 18 deepdefense.py:486] ratio : 0.224275 I0312 12:07:00.693285 18 deepdefense.py:487] ce_loss : 0.020267 I0312 12:07:00.693365 18 deepdefense.py:488] loss : 0.394605 I0312 12:07:00.693444 18 deepdefense.py:489] accuracy : 0.990000 I0312 12:07:00.766579 18 deepdefense.py:483] Processing 2700 - 2800 / 60000 I0312 12:07:00.766695 18 deepdefense.py:484] noise_norm : 424.162316 I0312 12:07:00.766824 18 deepdefense.py:485] grad_norm : 28.656344 I0312 12:07:00.766938 18 deepdefense.py:486] ratio : 0.227037 I0312 12:07:00.767028 18 deepdefense.py:487] ce_loss : 0.009213 I0312 12:07:00.767117 18 deepdefense.py:488] loss : 0.559985 I0312 12:07:00.767210 18 deepdefense.py:489] accuracy : 1.000000 I0312 12:07:00.846417 18 deepdefense.py:483] Processing 2800 - 2900 / 60000 I0312 12:07:00.846539 18 deepdefense.py:484] noise_norm : 441.390101 I0312 12:07:00.846654 18 deepdefense.py:485] grad_norm : 24.307082 I0312 12:07:00.846759 18 deepdefense.py:486] ratio : 0.238342 I0312 12:07:00.846866 18 deepdefense.py:487] ce_loss : 0.029882 I0312 12:07:00.846957 18 deepdefense.py:488] loss : 0.309963 I0312 12:07:00.847044 18 deepdefense.py:489] accuracy : 0.990000 I0312 12:07:00.931418 18 deepdefense.py:483] Processing 2900 - 3000 / 60000 I0312 12:07:00.931542 18 deepdefense.py:484] noise_norm : 391.050166 I0312 12:07:00.931686 18 deepdefense.py:485] grad_norm : 20.951988 I0312 12:07:00.931795 18 deepdefense.py:486] ratio : 0.210398 I0312 12:07:00.931886 18 deepdefense.py:487] ce_loss : 0.064489 I0312 12:07:00.931974 18 deepdefense.py:488] loss : 0.415758 I0312 12:07:00.932064 18 deepdefense.py:489] accuracy : 0.980000 ..... 这些结果如何保存？以便用于作图。
…
-- At 2021-03-12 09:25:41, "yanziang" @.***> wrote: Hi, all models will be saved in args.exp_dir. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

You can try to use > (shell) command to redirect stdout and stderr.

**Comment by aspnetcs** [comment]
deepdefense.pytorch is semi-supervise defense algorithm?













--




在 2021-03-12 22:02:54，"yanziang" ***@***.***> 写道：

Closed #5.

**Comment by ZiangYan** [comment]
> deepdefense.pytorch is semi-supervise defense algorithm?
> […](#)
> -- 在 2021-03-12 22:02:54，"yanziang" ***@***.***> 写道： Closed #5. — You are receiving this because you authored the thread. Reply to this email directly, view it on GitHub, or unsubscribe.

No.
